app/__init__.py

```diff
@@ -275,9 +275,12 @@ def create_random_identifier():
 def process_user_agent(user_agent_string):
     if user_agent_string and user_agent_string.lower().startswith("notify"):
         components = user_agent_string.split("/")
-        client_name = components[0].lower()
-        client_version = components[1].replace(".", "-")
-        return "{}.{}".format(client_name, client_version)
+        try:
+            client_name = components[0].lower()
+            client_version = components[1].replace(".", "-")
+            return "{}.{}".format(client_name, client_version)
```

Comment:
This code was meant to handle this case

```py
process_user_agent("NOTIFY-API-PYTHON-CLIENT/3.0.0")
# "notify-api-python-client.3-0-0"
```

It logs libs usage on StatsD

Reply:
I like this approach, if I can make a suggestion of using regex and making the matching a bit more strict. something like this:
`
def process_user_agent(agent_str: str):
    try:
        m = re.search(r'(?P<name>notify.*)/(?P<version>\d*.\d*.\d)', agent_str, re.IGNORECASE)
        return f'{m.group("name")}.{m.group("version").replace(".", "-")}' if m else 'non-notify-user-agent'
    except TypeError:
        return 'unknown'
`

Reply:
Refactored to regex in next commit.